src/algebra/char_zero.lean

```diff
@@ -53,6 +53,11 @@ instance linear_ordered_semiring.to_char_zero {R : Type*}
   [linear_ordered_semiring R] : char_zero R :=
 ⟨nat.strict_mono_cast.injective⟩
 
+/-- An `ordered_semiring` has characteristic zero. -/
+lemma char_zero_ordered_semiring {R : Type*} [ordered_semiring R] [nontrivial R] :
+  char_zero R :=
+{ cast_injective := nat.strict_mono_cast.injective }
+
```

Comment:
I think you should edit the 5 lines above too (which I can't do in a suggestion, to:
```lean
/-- Note this is not an instances as `char_zero` implies `nontrivial`, and this would risk forming a loop. -/
lemma ordered_semiring.to_char_zero {R : Type*} [ordered_semiring R] [nontrivial R] :
  char_zero R :=
⟨nat.strict_mono_cast.injective⟩

@[priority 100] -- see Note [lower instance priority]
instance linear_ordered_semiring.to_char_zero {R : Type*}
  [linear_ordered_semiring R] : char_zero R :=
ordered_semiring.to_char_zero
```

Reply:
Done, thanks!